DQM/SiPixelPhase1Heterogeneous/plugins/SiPixelPhase1MonitorClusters.cc

```diff
@@ -0,0 +1,385 @@
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DataFormats/SiPixelDigi/interface/SiPixelDigisSoA.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+
+class SiPixelPhase1MonitorClusters : public DQMEDAnalyzer {
+public:
+  explicit SiPixelPhase1MonitorClusters(const edm::ParameterSet&);
+  ~SiPixelPhase1MonitorClusters() override = default;
+  void bookHistograms(DQMStore::IBooker& ibooker, edm::Run const& iRun, edm::EventSetup const& iSetup) override;
+  void analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  const edm::EDGetTokenT<SiPixelClusterCollectionNew> tokenCluster_;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> trackerTopoToken_;
+  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> trackerGeomToken_;
+  std::string topFolderName_;
+  const TrackerTopology* trackerTopology_;
+
+  MonitorElement* hnClusters;
+  MonitorElement* hClustersCharge;
+  MonitorElement* hClustersSize;
+
+  MonitorElement* hnClustersBarrel;
+  MonitorElement* hClustersBarrelCharge;
+  MonitorElement* hClustersBarrelSize;
+
+  MonitorElement* hnClustersEndcap;
+  MonitorElement* hClustersEndcapCharge;
+  MonitorElement* hClustersEndcapSize;
+
+  MonitorElement* hnClustersBarrelLayer1;
+  MonitorElement* hnClustersBarrelLayer2;
+  MonitorElement* hnClustersBarrelLayer3;
+  MonitorElement* hnClustersBarrelLayer4;
+
+  MonitorElement* hClustersBarrelLayer1Charge;
+  MonitorElement* hClustersBarrelLayer2Charge;
+  MonitorElement* hClustersBarrelLayer3Charge;
+  MonitorElement* hClustersBarrelLayer4Charge;
+
+  MonitorElement* hClustersBarrelLayer1Size;
+  MonitorElement* hClustersBarrelLayer2Size;
+  MonitorElement* hClustersBarrelLayer3Size;
+  MonitorElement* hClustersBarrelLayer4Size;
+
+  MonitorElement* hnClustersEndcapDiskm1;
+  MonitorElement* hnClustersEndcapDiskm2;
+  MonitorElement* hnClustersEndcapDiskm3;
+  MonitorElement* hnClustersEndcapDiskp1;
+  MonitorElement* hnClustersEndcapDiskp2;
+  MonitorElement* hnClustersEndcapDiskp3;
+
+  MonitorElement* hClustersEndcapDiskm1Charge;
+  MonitorElement* hClustersEndcapDiskm2Charge;
+  MonitorElement* hClustersEndcapDiskm3Charge;
+  MonitorElement* hClustersEndcapDiskp1Charge;
+  MonitorElement* hClustersEndcapDiskp2Charge;
+  MonitorElement* hClustersEndcapDiskp3Charge;
+
+  MonitorElement* hClustersEndcapDiskm1Size;
+  MonitorElement* hClustersEndcapDiskm2Size;
+  MonitorElement* hClustersEndcapDiskm3Size;
+  MonitorElement* hClustersEndcapDiskp1Size;
+  MonitorElement* hClustersEndcapDiskp2Size;
+  MonitorElement* hClustersEndcapDiskp3Size;
+};
+
+//
+// constructors
+//
+
+SiPixelPhase1MonitorClusters::SiPixelPhase1MonitorClusters(const edm::ParameterSet& iConfig) {
+  tokenCluster_ = consumes<SiPixelClusterCollectionNew>(iConfig.getParameter<edm::InputTag>("pixelClusters"));
```

Comment:
now that they are declared `const` these should be in the class initializer list.